### PR TITLE
Update Bitwarden extension to add TOTP

### PR DESCRIPTION
**Pull request by jradtilbrook**
## Description

This PR adds TOTP fetching to the bitwarden extension.

## Type of change

- Improvement for somebody else's extension. Extension author: @pomdtr 

## Screenshot

Screen Shot 2021-12-10 at 12 32 53

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/bitwarden/src/search.tsx [hunk 1]
```diff
@@ -145,6 +153,16 @@ function ItemListItem(props: { item: Item; folder: Folder | undefined; refreshIt
 
   const tree = treeify.asTree(cleanItem, true, false);
 
+  async function copyTotp(id: string) {
```
Comment (pomdtr) [inline extensions/bitwarden/src/search.tsx:156]: Why not adding all the logic into the loadTotp function ? I feel like splitting the logic between the two functions make the code harder to grasp.

File: extensions/bitwarden/src/search.tsx [hunk 2]
```diff
@@ -61,6 +63,10 @@ function ItemList(props: {
     }
   }
 
+  function loadTotp(sessionToken: string) {
```
Comment (pomdtr) [inline extensions/bitwarden/src/search.tsx:66]: We could rename this to copyTOTP instead, and group all of the logic here.

**Review by pomdtr — APPROVED** [review]
Thanks for the contribution ! I added some quick feedback on the code, please consider them before merging the PR :)

**Comment by pernielsentikaer** [comment]
Hi @jradtilbrook 👋 

Could you look into the feedback from @pomdtr?

File: extensions/bitwarden/src/search.tsx (hunk 1 above)
Comment (jradtilbrook) [reply extensions/bitwarden/src/search.tsx:156]: Thanks @pomdtr - I agree! I haven't written react for a few years now so I wasn't sure of any security issues with passing the session token as a prop, so I tried to remove that by wrapping it in a closure. But if its fine to do, grouping it into one function is definitely better IMO

**Comment by jradtilbrook** [comment]
@pomdtr @pernielsentikaer Thanks for the review, I've refactored to one function now. Let me know what you think and if you think there's more that could be improved! 👍

**Comment by pernielsentikaer** [comment]
@pomdtr - I don't have BitWarden premium, do you? Can you check a last time? I'm not sure who should make it, but I'm missing a README

**Comment by pomdtr** [comment]
I do not have bitwarden premium either. Code looks fine to me.

**Pull request merged**